### DISCUSSION TITLE
migrations, knex

### DISCUSSION THREAD

**Discussion by korneliusz-w** [directus/directus#6647] [Archive]
HI

I want to use migration files to create database schema. I know how to create basic table with fileds but how to define interface for fields: Textfield -> WYSWIG, Json field -> Repeter, dropdown field ?
Also how to define fields relations  M2M or M2A ? 

?

**Comment by korneliusz-w** [comment]
Ok I found this post https://github.com/directus/directus/issues/6432
So I need to create new record to directus_fields table.

**Comment by korneliusz-w** [comment]
Today the commands up and down not working. Only latest

Error: Couldn't find migration

**Comment by korneliusz-w** [comment]
I removerd records from directus_migrations table and it works now.

**Comment by korneliusz-w** [comment]
No, still not working correctly :(

**Comment by rijkvanzanten** [comment]
The custom migrations extension (https://docs.directus.io/guides/migrations/) is very open ended, and basically allows you direct access to the database. In order to setup the fields you're asking for, you'll have to setup the records in directus_collections/fields/relations yourself. It might be easier to setup the fields in directus first, and then make a database dump to use as a starting point 👍🏻

**Reply by korneliusz-w** [reply]
And what is a difference between :latest and :up ?

**Reply by rijkvanzanten** [reply]
Up executes the next single migration, latest keeps running up until there's no newer migrations

**Reply by korneliusz-w** [reply]
@rijkvanzanten 

OK
Now I try to run :down migration but it not delete table.

npx directus database migrate:down
✨ Running migrations...
✨ Undoing Replace Groups...
✨ Downgrade successful

```js
const tableName = 'articles';

module.exports = {
    async up(knex) {
        await knex.schema.createTable(tableName, (table) => {
            table.increments('id').primary();
            table.string('status');
            table.datetime('date_created');
            table.string('title');
            table.string('introduction_text');
            table.text('highlighted_text');
            table.text('text', 'longtext');
            table.string('hero_image');
            table.boolean('is_seo');
            table.string('url');
        });
    },

    async down(knex) {
        await knex.schema.dropTable(tableName);
    },
};
```

**Reply by rijkvanzanten** [reply]
I'm assuming that the bit of code you pasted in is your custom migration. `npx directus database migrate:down` will downgrade the last ran migration, in this case the "Replace Groups" migration. That wasn't your custom migration by the looks of it. Try running `npx directus database migrate:down` again 👍🏻
